File: Case-study/Internet Service/FE/internet-service-management/src/app/interface/Employee.ts
import {IAccount} from "./IAccount";
import {IPosition} from "./IPosition";

export class Employee {
  private employeeId!: string;
  private fullName!: string;
  private dateOfBirth!: string;
  private email!: string;
  private address!: string;
  private phone!: string;
  private level!: string;
  private startWorkDate!: string;
  private yearOfExp!: number;
  private avtUrl!: string;
  private userName!: string;
  private positionId!: number;
  private password!: string;


  constructor(employeeId: string, fullName: string, dateOfBirth: string, email: string, address: string, phone: string, level: string, startWorkDate: string, yearOfExp: number, avtUrl: string, positionId: number, userName: string, password: string) {
    this.employeeId = employeeId;
    this.fullName = fullName;
    this.dateOfBirth = dateOfBirth;
    this.email = email;
    this.address = address;
    this.phone = phone;
    this.level = level;
    this.startWorkDate = startWorkDate;
    this.yearOfExp = yearOfExp;
    this.avtUrl = avtUrl;
    this.positionId = positionId;
    this.userName = userName;
    this.password = password;
  }
}

File: Case-study/Internet Service/BE/InternetServiceManagement/src/main/java/com/example/demo/entity/Employee.java
package com.example.demo.entity;

import javax.persistence.*;

@Entity
public class Employee {
    @Id
    private String employeeId;
    private String fullName;
    private String dateOfBirth;
    private String email;
    private String address;
    private String phone;
    private String level;
    private String startWorkDate;
    private int yearOfExp;
    private String avtUrl;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "userName",referencedColumnName = "userName")
    private Account account;

    @ManyToOne(targetEntity = Position.class)
    @JoinColumn(name = "positionId", referencedColumnName = "positionId")
    private Position position;

    public Employee() {
    }

    public Employee(String employeeId, String fullName, String dateOfBirth, String email, String address, String phone, String level, String startWorkDate, int yearOfExp, String avtUrl, Account account, Position position) {
        this.employeeId = employeeId;
        this.fullName = fullName;
        this.dateOfBirth = dateOfBirth;
        this.email = email;
        this.address = address;
        this.phone = phone;
        this.level = level;
        this.startWorkDate = startWorkDate;
        this.yearOfExp = yearOfExp;
        this.avtUrl = avtUrl;
        this.account = account;
        this.position = position;
    }

    public String getEmployeeId() {
        return employeeId;
    }

    public void setEmployeeId(String employeeId) {
        this.employeeId = employeeId;
    }

    public String getFullName() {
        return fullName;
    }

    public void setFullName(String fullName) {
        this.fullName = fullName;
    }

    public String getDateOfBirth() {
        return dateOfBirth;
    }

    public void setDateOfBirth(String dateOfBirth) {
        this.dateOfBirth = dateOfBirth;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getLevel() {
        return level;
    }

    public void setLevel(String level) {
        this.level = level;
    }

    public String getStartWorkDate() {
        return startWorkDate;
    }

    public void setStartWorkDate(String startWorkDate) {
        this.startWorkDate = startWorkDate;
    }

    public int getYearOfExp() {
        return yearOfExp;
    }

    public void setYearOfExp(int yearOfExp) {
        this.yearOfExp = yearOfExp;
    }

    public String getAvtUrl() {
        return avtUrl;
    }

    public void setAvtUrl(String avtUrl) {
        this.avtUrl = avtUrl;
    }

    public Account getAccount() {
        return account;
    }

    public void setAccount(Account account) {
        this.account = account;
    }

    public Position getPosition() {
        return position;
    }

    public void setPosition(Position position) {
        this.position = position;
    }
}
